### PR TITLE
Allow setting MTU for calico networking.

### LINKED ISSUE #4042: Expose the Calico interface and the ipip tunnel MTU for configuration in the kops specs
Thanks for submitting an issue! Please fill in as much of the template below as
you can.

------------- FEATURE REQUEST TEMPLATE --------------------

1. Describe IN DETAIL the feature/behavior/change you would like to see.

Please expose the mtu variables for Calico, both for calico interface and the ipip tunnel, similar to what is done for weave:

```
spec:
  networking:
    weave:
      mtu: 8912
```

2. Feel free to provide a design supporting your feature request.

### PR DESCRIPTION
**Pull request by shrinandj**
This commit does the following two changes:

1. Changes the default calico mtu to 8198.
2. Enables setting the mtu explicitly in the config as:
```
  networking:
    calico:
      mtu: 2048
```

Testing done:

1. Created cluster on AWS with networking set to calico. No additional details were provided. Verified that the mtu was set to 8198. Also verified that the FELIX_IPINIPMTU environment variable was set to 8198.
2. Created a cluster explicitly setting the calico mtu to 2048. Verified that the mtu for the 'cali*' interfaces inside the pods was set to 2048. Also, verified that the FELIX_IPINIPMTU environment variable was set to 2048.
3. make test passed.

Closes #4042

### CODE REVIEW

File: pkg/apis/kops/networking.go [hunk 1]
```diff
@@ -83,6 +83,8 @@ type CalicoNetworkingSpec struct {
 	PrometheusGoMetricsEnabled bool `json:"prometheusGoMetricsEnabled,omitempty"`
 	// PrometheusProcessMetricsEnabled enables Prometheus process metrics collection
 	PrometheusProcessMetricsEnabled bool `json:"prometheusProcessMetricsEnabled,omitempty"`
+	// MTU to be set in the cni-network-config for calico.
+	MTU *int32 `json:"mtu,omitempty"`
```
Comment (gambol99) [inline pkg/apis/kops/networking.go]: Can this be added in alphabetically order please ..

File: upup/models/cloudup/resources/addons/networking.projectcalico.org/k8s-1.7.yaml.template [hunk 2]
```diff
@@ -216,6 +217,8 @@ spec:
               value: "{{- or .Networking.Calico.PrometheusProcessMetricsEnabled "true" }}"
             - name: FELIX_HEALTHENABLED
               value: "true"
+            - name: FELIX_IPINIPMTU
```
Comment (gambol99) [inline upup/models/cloudup/resources/addons/networking.projectcalico.org/k8s-1.7.yaml.template]: What is the default if not set? ... Also, the [docs](https://docs.projectcalico.org/v3.0/usage/configuration/mtu) indicate encapsulation and cloud provider makes a difference

File: pkg/apis/kops/networking.go (hunk 1 above)
Comment (shrinandj) [reply pkg/apis/kops/networking.go]: Done.

File: upup/models/cloudup/resources/addons/networking.projectcalico.org/k8s-1.7.yaml.template (hunk 2 above)
Comment (shrinandj) [reply upup/models/cloudup/resources/addons/networking.projectcalico.org/k8s-1.7.yaml.template]: The default is 8918 if not set. Users can set it to the correct option based on encapsulation and cloud provider.

**Comment by justinsb** [comment]
Thanks for the contribution @shrinandj!  I think it's a great idea to expose setting the MTU, what gives me pause is forcing the change on users.

One option is to do what weave does - if the user specifies the MTU we pass it through to calico, otherwise we let calico default it.  This is good because Calico should be able to figure out an optimal MTU.  It also means that we don't need to worry about the idea that different clouds might allow different packet sizes.

If you really want to set the default, one way to do it is to do it only for new clusters.  So in the create_cluster.go flow, you would default the calico MTU to whatever default value you want.  This has the downside that the value then appears in everyone's manifests.

I'm fine with this if we remove the defaulting, but it's very tricky to default the value for everyone.  Would you be OK with removing the defaulting, so it behaves the same way as weave?

**Comment by shrinandj** [comment]
@justinsb The latest commit implements your suggestion. The values will be overridden only if explicitly set in the kops cluster config. Otherwise, the behavior will be exactly as it was earlier.

Let me know what you'll think about this.

**Comment by shrinandj** [comment]
It would be really good if this could go into the next `kops` release. Let me know if there are any more suggestions/changes required to get this in.

**Comment by justinsb** [comment]
Thanks @shrinandj

/approve
/lgtm

**Pull request merged**